File: 1449-print-words-vertically/print-words-vertically.py
class Solution(object):
    def printVertically(self, s):
        l = max([len(word) for word in s.split()])
        store = [[] for _ in range(l)]

        for word in s.split():
            for index in range(l):
                if index < len(word):
                    store[index].append(word[index])
                else:
                    store[index].append(" ")

        res = ["".join(chars) for chars in store]
        store = []

        for word in res:
            lenght = len(word) - 1
            while lenght and word[lenght] == " ":
                lenght -= 1
            store.append(word[:lenght + 1])

        return store

File: 1449-print-words-vertically/print-words-vertically.go
func printVertically(s string) []string {

    var result []string
    words := strings.Split(s, " ")

    maxLength := 0

    for _, word := range words{
        if maxLength < len(word){
            maxLength = len(word)
        }
    }

    for i := 0 ; i < maxLength; i++{
        var verticalWord []string

        for j := 0; j < len(words); j ++{
            if i < len(words[j]){
                verticalWord = append(verticalWord,string( words[j][i]))
            } else {
                verticalWord = append(verticalWord, " ")
            }
        }

        result = append(result, strings.TrimRight(strings.Join(verticalWord, ""), " "))
    }

    return result
    
}